java/src/com/ibm/streamsx/rest/Element.java

```diff
@@ -37,4 +42,31 @@ void setConnection(StreamsConnection connection) {
     public final String toString() {
         return pretty.toJson(this);
     }
+    
+    private Gson refreshJson;
+    
+    /**
+     * Refresh this element.
+     * 
+     * Attributes of this object are updated from current state
+     * though the Streams REST API.
+     * 
+     * @throws IOException Error communicating with Streams.
+     */
+    public void refresh() throws IOException {
+        String response = connection().getResponseString(self);
```

Comment:
oddly enough, the Metrics object which is one that we'd like the refesh() to work on:
https://www.ibm.com/support/knowledgecenter/en/SSCRJU_4.2.1/com.ibm.streams.restapi.doc/doc/restapis-metrics.html

based on what I'd seen, the only way to refresh the metrics objects will be to call the getMetrics() from the higher level object (i.e, Operator,  the Input/Output Ports, etc, ..)